typescript/packages/wallets/starknet/src/StarknetWalletClient.ts

```diff
@@ -0,0 +1,112 @@
+import { WalletClientBase } from "@goat-sdk/core";
+import { formatUnits } from "ethers";
+import type { Account, Call, InvocationsDetails, InvokeFunctionResponse, RpcProvider } from "starknet";
+
+export type StarknetWalletCtorParams = {
+    starknetAccount: Account;
+    starknetClient: RpcProvider;
+};
+
+export type StarknetTransaction = {
+    calls: Call[];
+    transactionDetails?: InvocationsDetails;
+};
+
+export class StarknetWalletClient extends WalletClientBase {
```

Comment:
I would make this an abstract class so other types of wallets (custodial, smart, etc.) can extend it. Checkout the solana or sui implementations: 
- https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/solana/src/SolanaWalletClient.ts
- https://github.com/goat-sdk/goat/blob/main/typescript/packages/wallets/sui/src/SuiWalletClient.ts